packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1076,13 +1089,87 @@ class RenderParagraph extends RenderBox
           buffer.write(label);
           offset += label.length;
         }
-        _cachedAttributedLabel = AttributedString(buffer.toString(), attributes: attributes);
+        _cachedAttributedLabels = <AttributedString>[AttributedString(buffer.toString(), attributes: attributes)];
       }
-      config.attributedLabel = _cachedAttributedLabel!;
+      config.attributedLabel = _cachedAttributedLabels![0];
       config.textDirection = textDirection;
     }
   }
 
+  ChildSemanticsConfigurationsResult _childSemanticsConfigurationsDelegate(List<SemanticsConfiguration> childConfigs) {
+    final List<AttributedString> cachedStrings = _cachedAttributedLabels ??= <AttributedString>[];
+    final ChildSemanticsConfigurationsResultBuilder builder = ChildSemanticsConfigurationsResultBuilder();
+    final bool hasCache = cachedStrings.isNotEmpty;
+    int placeholderIndex = 0;
+    int childConfigsIndex = 0;
+    int textSpanLabelIndex = 0;
+    bool seenTextSpan = false;
+    int offset = 0;
+    StringBuffer? buffer;
+    List<StringAttribute>? attributes;
+    for (final InlineSpanSemanticsInformation info in _semanticsInfo!) {
+      if (info.isPlaceholder) {
+        if (seenTextSpan) {
+          final SemanticsConfiguration textSpanConfig = SemanticsConfiguration();
+          textSpanConfig.textDirection = textDirection;
+          if (hasCache) {
+            textSpanConfig.attributedLabel = cachedStrings[textSpanLabelIndex];
+          } else {
+            assert(buffer != null && attributes != null);
+            textSpanConfig.attributedLabel = AttributedString(buffer.toString(), attributes: attributes!);
+            assert(textSpanLabelIndex == cachedStrings.length);
+            cachedStrings.add(textSpanConfig.attributedLabel);
+            buffer = null;
+            offset = 0;
+            attributes = null;
+          }
+          textSpanLabelIndex += 1;
+          builder.markAsMergeUp(textSpanConfig);
+          seenTextSpan = false;
+        }
+        while (childConfigsIndex < childConfigs.length &&
+            childConfigs[childConfigsIndex].tagsChildrenWith(PlaceholderSpanIndexSemanticsTag(placeholderIndex))) {
+          builder.markAsMergeUp(childConfigs[childConfigsIndex]);
+          childConfigsIndex += 1;
+        }
+        placeholderIndex += 1;
+        continue;
+      }
+      seenTextSpan = true;
+      if (!hasCache) {
```

Comment:
I may not fully understand this logic, yet, but I would have expected that this branch would put something in the cache? And then there would be an else branch pulling it out of the cache again?